### ISSUE TITLE
Crash when closing tab

### ISSUE THREAD

**Issue by sokoAwen** [Dpeta/pesterchum-alt-servers#36]
When I have multiple tabs open and try to close one I get a crash:
`Traceback (most recent call last):
  File "convo.py", line 280, in changeTab
`
I am on Debian 11, QT 5.15.2. Hopefully I've provided adequate info. :]

**Comment by Dpeta** [comment]
Hmmm, I'm not sure. I'm also on Debian 11 but I don't seem to be able to reproduce the error :0c

**Comment by Dpeta** [comment]
I think I was able to reproduce the crash, though, I got a completely different error. 
```Traceback (most recent call last):
  File "/home/XXX/Desktop/pesterchum-alt-servers/profile.py", line 58, in log
  File "/usr/lib/python3.9/codecs.py", line 905, in open
OSError: [Errno 24] Too many open files: '/home/XXX/.pesterchum/logs/XXX/XXX/bbcode/XXX.2021-07-22.05.22.txt'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/XXX/Desktop/pesterchum-alt-servers/pesterchum.py", line 2084, in pesterSelectedChum
  File "/home/XXX/Desktop/pesterchum-alt-servers/pesterchum.py", line 2092, in newConversationWindow
  File "/home/XXX/Desktop/pesterchum-alt-servers/pesterchum.py", line 1628, in newConversation
  File "/home/XXX/Desktop/pesterchum-alt-servers/convo.py", line 661, in __init__
  File "/home/XXX/Desktop/pesterchum-alt-servers/profile.py", line 60, in log
TypeError: arguments did not match any overloaded call:
  QMessageBox(parent: QWidget = None): argument 1 has unexpected type 'PesterLog'
  QMessageBox(QMessageBox.Icon, str, str, buttons: Union[QMessageBox.StandardButtons, QMessageBox.StandardButton] = QMessageBox.NoButton, parent: QWidget = None, flags: Union[Qt.WindowFlags, Qt.WindowType] = Qt.Dialog|Qt.MSWindowsFixedSizeDialogHint): argument 1 has unexpected type 'PesterLog'
Aborted```

Here it seems to be an IO error, but, I'm not certain if increasing the ulimit is the correct way to go about this?

**Comment by Dpeta** [comment]
I don't think increasing the ulimit is even an option for most operating systems, actually-
I think it's caused by a kernel-imposed limit on how many files a program is allowed to have open at the same time, and Pesterchum seems to have 3 logs files open for every tab-
Maybe opening it and then closing it when writing would be a better way to go about it? That doesn't seem like it'll be an easy change though.

**Comment by Dpeta** [comment]
Actually, upon further inspection, the issue isn't necessarily logs at all- It's the emotes :0c
For every open tab Pesterchum seems to re-open every /smilies/ file and keep it opened, which brings us close to the cap pretty quickly

**Comment by Dpeta** [comment]
Okay well, the emote handle leak is probably fixed now- ([a82d460](https://github.com/Dpeta/pesterchum-alt-servers/commit/a82d460da4aefcd97e0e7e560570e6ad07407daf))

That probably wasn't your issue though, it's hard to tell though since that's definitely not the full error message, it has the relevant function but not actually the exception :"3

**Comment by Dpeta** [comment]
Can't reproduce this so not sure if it still happens, closing for now. :0c
